tests/L0/run_transformer/run_bert_minimal_test.py

```diff
@@ -0,0 +1,168 @@
+  temps = list()
+  for i in range(batch_size):
+     if inds is None or data_idx >= len(inds):
+       # hack as use of RNG will fall out of sync due to pipelines being different
+       torch.manual_seed(MANUAL_SEED)
+       inds = torch.randperm(effective_length, device='cuda')
+       masks = (torch.rand(len(inds)//batch_size + 1, batch_size, sequence_len, device='cuda') >= MASK_PROB).long()
+       MANUAL_SEED += 1
+       print("new epoch", len(inds))
+       data_idx = 0
+       print("my start", inds[0:5])
+       print("masks_checksum:", torch.sum(masks))
+     if EASY_MODE:
+       data_idx_ = data_idx % EASY_MODE_SIZ
+     else:
+       data_idx_ = data_idx
+     offset = inds[data_idx_] #* SEQUENCE_LEN
+     data_idx += 1
+      
+     curr = fancy_data[offset:offset+sequence_len].clone().detach()
+     temps.append(curr)
+  temp = torch.stack(temps, dim=0).cuda()
+  mask = masks[data_idx//batch_size]
+  mask_not = torch.logical_not(mask)
+  data = mask * temp + mask_not*124
+  label = temp
+  return (data, label, mask_not)
+
+easy_data = None
+
+def fwd_step_func(batch, model):
+    global ONCE
+    data, label, loss_mask = batch
+    data = data.cuda()
+    label = label.cuda()
+    loss_mask = loss_mask.cuda()
+    y = model(data, torch.ones_like(data), lm_labels=label)
+
+    def loss_func(output_tensor):
+        output_tensor, _ = output_tensor
+        lm_loss_ = output_tensor.float()
+        lm_loss = torch.sum(
+                        lm_loss_.view(-1) * loss_mask.reshape(-1)) / loss_mask.sum()
+        averaged_loss = average_losses_across_data_parallel_group([lm_loss])
+        if data_idx >= 1536:
+            assert lm_loss < 4.8
+            if not ONCE:
+                print("LOSS OK")
+                ONCE = True
+        return lm_loss, {'avg': averaged_loss}
+    return y, loss_func
+
+
+def train(model, optim):
+    sequence_len = global_vars.get_args().seq_length
+    micro_batch_size = global_vars.get_args().micro_batch_size
+    hidden_size = global_vars.get_args().hidden_size
+    forward_backward_func = _forward_backward_pipelining_with_interleaving
+    tensor_shape = (args.seq_length, args.micro_batch_size, args.hidden_size)
+    for _ in range(8):
+        batch = generate_fancy_data_labels(sequence_len, batch_size)
+        optim.zero_grad()
+        forward_backward_func(fwd_step_func, batch, model, forward_only=False, tensor_shape=tensor_shape)
+        optim.step()
+
+if __name__ == '__main__':
+    global fancy_data
+    global effective_length
+
+    global_vars.set_global_variables()
+
+    fancy_data = download_fancy_data()
+    effective_length = fancy_data.size(0) // global_vars.get_args().seq_length
+    effective_length = fancy_data.size(0) - global_vars.get_args().seq_length
+
+
+    initialize_distributed()
+    world_size = torch.distributed.get_world_size()
+    failure = None
+    try:
+        args = global_vars.get_args()
+        args.padded_vocab_size = 128 # needed in standalone gpt
```

Comment:
[nit-picking] is this `padded_vocab_size` needed in Bert? If so, could you set this value via `global_vars.set_global_variables()` like https://github.com/NVIDIA/apex/blob/9b7d2ac080d7aaa3d44622b8b0cd7a0299e43ea5/tests/L0/run_transformer/run_dynamic_batchsize_test.py#L33-L36?

Reply:
It looks like this requires the argument to already be defined; here the hack is because we're skipping the tokenizer which usually sets it: https://github.com/NVIDIA/Megatron-LM/blob/42c1cf4279acea5a554500dcb552211f44cbec45/megatron/tokenizer/tokenizer.py#L25

Reply:
okay.